src/dd/dd.rs

```diff
@@ -0,0 +1,679 @@
+        result
+    }
+
+    const BLOCK: FlagT = 0b00000001;
+    const UNBLOCK: FlagT = 0b00000010;
+    const LCASE: FlagT = 0b00000100;
+    const UCASE: FlagT = 0b00001000;
+    const SWAB: FlagT = 0b00010000;
+    const NOERROR: FlagT = 0b00100000;
+    const NOTRUNC: FlagT = 0b01000000;
+    const SYNC: FlagT = 0b10000000;
+    const INVALID_FLAG: FlagT = 0b1_00000000;
+}
+
+impl fmt::Display for Conf {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "if: {input_filename}\n\
+             of: {output_filename}\n\
+             ibs: {ibs}\n\
+             obs: {obs}\n\
+             cbs: {cbs}\n\
+             skip: {skip}\n\
+             seek: {seek}\n\
+             count: {count}\n\
+             flags: {flags}\n",
+            input_filename = self.inp_name,
+            output_filename = self.out_name,
+            ibs = self.inp_block_size,
+            obs = self.out_block_size,
+            cbs = self.cnv_block_size,
+            skip = self.skip,
+            seek = self.seek,
+            count = self.count,
+            flags = self.str_flags()
+        )
+    }
+}
+
+pub fn parse_args() -> Conf {
+    let mut cf = Conf::default();
+    for arg in env::args().skip(1) {
+        let pair: Vec<&str> = arg.splitn(2, '=').collect();
+        if pair.len() == 2 {
+            let (key, val) = (pair[0], pair[1]);
+            match key {
+                "if" => cf.set_if(val),
+                "of" => cf.set_of(val),
+                "conv" => for elem in val.split(",") {
+                    match Conf::get_flag(elem) {
+                        Conf::INVALID_FLAG => err("invalid flag passed to conv"),
+                        flag => cf.set_flag(flag),
+                    }
+                },
+                _ => {
+                    if Conf::is_flag_valid(key) {
+                        if let Some(m) = get_u(val) {
+                            match key {
+                                "ibs" => cf.inp_block_size = m,
+                                "obs" => cf.out_block_size = m,
+                                "bs" => {
+                                    cf.inp_block_size = m;
+                                    cf.out_block_size = m;
+                                }
+                                "cbs" => cf.cnv_block_size = m,
+                                "skip" => cf.skip = m,
+                                "seek" => cf.seek = m,
+                                "count" => cf.count = m,
+                                _ => err(format!("invalid argument: {}", key)),
+                            }
+                        }
+                    } else {
+                        err(format!("invalid argument: {}", arg));
+                    }
+                }
+            }
+        } else {
+            err(format!("invalid argument: {}", arg));
+        }
+    }
+
+    // Behaviour is unspecified. We choose to terminate.
+    if cf.count == 0 {
```

Comment:
This causes the program to fail if you run it like `dd if=infile of=outfile`.  You probably want to wrap this in an `Option` with `None` being the default.  It should only fail if `cf.count == Some(0)`.

Reply:
Changed.